base/lock.jl

```diff
@@ -221,7 +221,7 @@ When this function returns, the `lock` has been released, so the caller should
 not attempt to `unlock` it.
 
 !!! compat "Julia 1.7"
-    Using a [`Channel`](@ref) as the second argument requires Julia 1.7 or later.
```

Comment:
Looks like this is actually correct, see https://github.com/JuliaLang/julia/commit/d9f0f04477bde51d858400a123cf40ccf4adc38d. Perhaps the best would be to update this docstring to a generic two-argument `lock` method?